### DISCUSSION TITLE
Not able to utilise simple task output value in DO_WHILE task loop condition

### DISCUSSION THREAD

**Discussion by shriram-apps** [Netflix/conductor#2419] [General]
**Describe**
In my current workflow i have created on DO_WHILE task which has multiple simple task inside the loopvars. I want to utilise couple of the simple task output values in the loopCondition which will help the workflow to process in loop to execute the simple task till the required data is received

**Describe the bug**
In my current workflow i have created on DO_WHILE task which has multiple simple task inside the loopvars. I want to utilise couple of the simple task output values in the loopCondition which will help the workflow to process in loop to execute the simple task till the required data is received

**Example:**
i have 2 simple tasks TASK A & TASK B both the tasks return the status value in OUTPUT which i am not able to access in loopcondition

Below is the workflow
curl -X POST http://localhost:8080/api/metadata/workflow -H 'Content-Type: application/json' -d '{
"name": "kotak_user_flow_loop",
"description": "Kotak user flow loop",
"version": 67,
"schemaVersion": 2,
"tasks": [
{
"name": "loop_task",
"taskReferenceName": "LoopTask",
"type": "DO_WHILE",
"inputParameters": {
"value": "${workflow.input.userId}"
},
"loopCondition": "if ( ($.kotak_nfdl_check_parallel.output.status === 'success' && $.kotak_ncif_check_parallel.output.status === 'success')) { true; } else { false; }",
"loopOver": [
{
"name": "fetch_user_details",
"taskReferenceName": "user_details",
"inputParameters": {
"userId": "${workflow.input.userId}"
},
"type": "SIMPLE"
},
{
"name": "parallel_process_nfcl_ncif",
"taskReferenceName": "parallel_nfcl_ncif",
"type": "FORK_JOIN",
"forkTasks": [
[
{
"name": "kotak_nfdl_check_new",
"taskReferenceName": "kotak_nfdl_check_parallel",
"inputParameters": {
"userId": "${user_details.output.panId}"
},
"type": "SIMPLE"
}
],
[
{
"name": "kotak_ncif_check",
"taskReferenceName": "kotak_ncif_check_parallel",
"inputParameters": {
"userId": "${user_details.output.pinCode}"
},
"type": "SIMPLE"
}
]
]
},
{
"name": "join_nfcl_ncif",
"taskReferenceName": "process_join_nfcl_ncif",
"type": "JOIN",
"joinOn": [
"kotak_nfdl_check_parallel",
"kotak_ncif_check_parallel"
]
},
{
"name": "kotak_user_dedupe_check",
"taskReferenceName": "kotak_user_dedupe",
"inputParameters": {
"nfdlStatus": "${kotak_nfdl_check_parallel.output.status}",
"ncifStatus": "${kotak_ncif_check_parallel.output.status}",
"userId": "${workflow.input.userId}",
},
"type": "SIMPLE"
}
],
"startDelay": 0,
"optional": false
}
]
}'

Note : i have also tried creating VARIABLE TASK and accessed it but no luck

Guide me how to access the task variable in DO_WHILE loop condition:
i have 2 simple tasks TASK A & TASK B both the tasks return the status value in OUTPUT which i am not able to access in loopcondition

Below is the workflow
curl -X POST http://localhost:8080/api/metadata/workflow -H 'Content-Type: application/json' -d '{
"name": "kotak_user_flow_loop",
"description": "Kotak user flow loop",
"version": 67,
"schemaVersion": 2,
"tasks": [
{
"name": "loop_task",
"taskReferenceName": "LoopTask",
"type": "DO_WHILE",
"inputParameters": {
"value": "${workflow.input.userId}"
},
"loopCondition": "if ( ($.kotak_nfdl_check_parallel.output.status === 'success' && $.kotak_ncif_check_parallel.output.status === 'success')) { true; } else { false; }",
"loopOver": [
{
"name": "fetch_user_details",
"taskReferenceName": "user_details",
"inputParameters": {
"userId": "${workflow.input.userId}"
},
"type": "SIMPLE"
},
{
"name": "parallel_process_nfcl_ncif",
"taskReferenceName": "parallel_nfcl_ncif",
"type": "FORK_JOIN",
"forkTasks": [
[
{
"name": "kotak_nfdl_check_new",
"taskReferenceName": "kotak_nfdl_check_parallel",
"inputParameters": {
"userId": "${user_details.output.panId}"
},
"type": "SIMPLE"
}
],
[
{
"name": "kotak_ncif_check",
"taskReferenceName": "kotak_ncif_check_parallel",
"inputParameters": {
"userId": "${user_details.output.pinCode}"
},
"type": "SIMPLE"
}
]
]
},
{
"name": "join_nfcl_ncif",
"taskReferenceName": "process_join_nfcl_ncif",
"type": "JOIN",
"joinOn": [
"kotak_nfdl_check_parallel",
"kotak_ncif_check_parallel"
]
},
{
"name": "kotak_user_dedupe_check",
"taskReferenceName": "kotak_user_dedupe",
"inputParameters": {
"nfdlStatus": "${kotak_nfdl_check_parallel.output.status}",
"ncifStatus": "${kotak_ncif_check_parallel.output.status}",
"userId": "${workflow.input.userId}",
},
"type": "SIMPLE"
}
],
"startDelay": 0,
"optional": false
}
]
}'

Note : i have also tried creating VARIABLE TASK and accessed it but no luck

Guide me how to access the task variable in DO_WHILE loop condition

**Comment by Jiehong** [comment]
@shriram-apps : did you have a look at the documentation for do_while? https://netflix.github.io/conductor/configuration/systask/#do-while-task

It shows how to access any iteration's outputs of the sub tasks.

**Reply by shriram-apps** [reply]
Thanks for the Proper URL

**Comment by aravindanr** [comment]
>Each task output is stored as part of the DO_WHILE task, indexed by the iteration value (see example below), allowing the condition to reference the output of a task for a specific iteration (eg. $.LoopTask['iteration]['first_task'])

@shriram-apps I believe this is what you are looking for
